src/wallet/scriptpubkeyman.cpp

```diff
@@ -1108,7 +1109,9 @@ void LegacyScriptPubKeyMan::ReturnDestination(int64_t nIndex, bool fInternal, co
         } else {
             setExternalKeyPool.insert(nIndex);
         }
-        m_pool_key_to_index[pubkey.GetID()] = nIndex;
+        CKeyID& pubkey_id = m_reserved_key_to_index.at(nIndex);
```

Comment:
Do we really want to crash if that entry doesn't exist?

I guess this is supposed to never happen, since `nIndex==-1` unless it was set otherwise in `GetReservedDestination` where `m_reserved_key_to_index` is also populated.